File: app/src/main/java/com/example/todonotesapp/DetailActivity.java
package com.example.todonotesapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

public class DetailActivity extends AppCompatActivity {
    TextView textViewTitle, textViewDescription;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        bindViews();
        setupIntentData();
    }

    private void setupIntentData() {
        Intent intent = getIntent();
        Log.d("DetailActivity", intent.getStringExtra(AppConstants.TITLE));
        String title = intent.getStringExtra(AppConstants.TITLE);
        String description = intent.getStringExtra(AppConstants.DESCRIPTION);
        textViewTitle.setText(title);
        textViewDescription.setText(description);

    }

    private void bindViews() {
        textViewTitle = findViewById(R.id.textViewTitle);
        textViewDescription = findViewById(R.id.textViewDescription);

    }
}

File: app/src/main/java/com/example/todonotesapp/view/DetailActivity.kt
package com.example.todonotesapp.view

import android.os.Bundle
import android.os.PersistableBundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.todonotesapp.utils.AppConstants
import com.example.todonotesapp.R

class DetailActivity:AppCompatActivity() {
    lateinit var textViewTitle:TextView
    lateinit var textViewDescription:TextView

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        setContentView(R.layout.activity_detail)

        bindViews()
        setupIntentData()
    }

    private fun bindViews() {
        textViewTitle = findViewById(R.id.textViewTitle)
        textViewDescription = findViewById(R.id.textViewDescription)
    }

    private fun setupIntentData() {
        val intent = intent;
        val title = intent.getStringExtra(AppConstants.TITLE)
        val description = intent.getStringExtra(AppConstants.DESCRIPTION)
        textViewTitle.text = title
        textViewDescription.text = description


    }

}